egs/swbd/s5c/local/nnet3/run_discriminative.sh

```diff
@@ -0,0 +1,221 @@
+
+  steps/nnet3/align.sh  --cmd "$decode_cmd $gpu_opts" --use-gpu "$use_gpu" \
+     --online-ivector-dir $online_ivector_dir \
+     --nj $nj $train_data_dir data/lang $srcdir ${srcdir}_ali || exit 1;
+
+fi
+
+if [ -z "$lats_dir" ]; then
+  lats_dir=${srcdir}_denlats
+  if [ $stage -le 2 ]; then
+    nj=50  # this doesn't really affect anything strongly, except the num-jobs for one of
+    # the phases of get_egs_discriminative2.sh below.
+    num_threads_denlats=6
+    subsplit=40 # number of jobs that run per job (but 2 run at a time, so total jobs is 80, giving
+    # total slots = 80 * 6 = 480.
+    steps/nnet3/make_denlats.sh --cmd "$decode_cmd --mem 1G --num-threads $num_threads_denlats" \
+      --online-ivector-dir $online_ivector_dir --determinize $determinize \
+      --nj $nj --sub-split $subsplit --num-threads "$num_threads_denlats" --config conf/decode.config \
+      $train_data_dir data/lang $srcdir ${lats_dir} || exit 1;
+  fi
+fi
+
+
+left_context=`nnet3-am-info $srcdir/final.mdl | grep "left-context:" | awk '{print $2}'` || exit 1
+right_context=`nnet3-am-info $srcdir/final.mdl | grep "right-context:" | awk '{print $2}'` || exit 1
+
+frame_subsampling_opt=
+if [ -f $srcdir/frame_subsampling_factor ]; then
+  frame_subsampling_opt="--frame-subsampling-factor $(cat $srcdir/frame_subsampling_factor)"
+fi
+
+cmvn_opts=`cat $srcdir/cmvn_opts` || exit 1
+
+if [ -z "$degs_dir" ]; then
+  degs_dir=${srcdir}_degs_n${frames_per_eg}_o${frames_overlap_per_eg}_f
+  if $determinize; then
+    degs_dir=${degs_dir}d
+  fi
+  if $minimize; then
+    degs_dir=${degs_dir}m
+  fi
+  if $remove_output_symbols; then
+    degs_dir=${degs_dir}r
+  fi
+  if $remove_epsilons; then
+    degs_dir=${degs_dir}e
+  fi
+  if $collapse_transition_ids; then
+    degs_dir=${degs_dir}c
+  fi
+
+  if [ $stage -le 3 ]; then
+    if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d ${srcdir}_degs/storage ]; then
+      utils/create_split_dir.pl \
+        /export/b0{1,2,5,6}/$USER/kaldi-data/egs/swbd-$(date +'%m_%d_%H_%M')/s5/${srcdir}_degs/storage ${srcdir}_degs/storage
+    fi
+    # have a higher maximum num-jobs if
+    if [ -d ${srcdir}_degs/storage ]; then max_jobs=10; else max_jobs=5; fi
+
+    degs_opts="--determinize $determinize --minimize $minimize --remove-output-symbols $remove_output_symbols --remove-epsilons $remove_epsilons --collapse-transition-ids $collapse_transition_ids"
+
+    steps/nnet3/get_egs_discriminative.sh \
+      --cmd "$decode_cmd --max-jobs-run $max_jobs --mem 20G" --stage $get_egs_stage --cmvn-opts "$cmvn_opts" \
+      --adjust-priors $adjust_priors \
+      --online-ivector-dir $online_ivector_dir --left-context $left_context --right-context $right_context $frame_subsampling_opt \
+      --criterion $criterion --frames-per-eg $frames_per_eg --frames-overlap-per-eg $frames_overlap_per_eg ${degs_opts} \
+      $train_data_dir data/lang ${srcdir}_ali $lats_dir $srcdir/final.mdl $degs_dir || exit 1;
+  fi
+fi
+
+d=`basename $degs_dir`
```

Comment:
again, please get rid of the mangled directory names and make it user configurable.
run.sh scripts are not supposed to be run with options except --stage and --train-stage.